### DISCUSSION TITLE
Error Using Server to Run Moose

### DISCUSSION THREAD

**Discussion by FanZhangchen** [idaholab/moose#24173] [Q&A General] [answered]
Hi Moose Team,

I updated the Moose and encountered problems when I tried to Makefile in Moose on the server. The error I got is shown below

`/moose/framework/src/base/MooseApp.C:401:48: error: 'class libMesh::Parallel::Communicator' has no member named 'sync_type'     
 401 |   const_cast<Parallel::Communicator &>(comm()).sync_type(timpi_sync);`

Could you give some suggestions to solve this problem?

Best,

zc

**Comment by GiudGiud** [comment, marked as answer]
Hello

You need to update libmesh.
`mamba update --all` if you used mamba,
OR
`scripts/update_and_rebuild_libmesh.sh` if you used the manual installation

Guillaume

**Reply by milljm** [reply]
When you update MOOSE, you need to update Mamba. When you update Mamba, you need to update MOOSE.

**Reply by FanZhangchen** [reply]
Hi,

I have already used `mamba update --all`. And it shows that no package needs to be updated. The info is given below

conda-forge/linux-64                                        Using cache
conda-forge/noarch                                          Using cache
pkgs/main/linux-64                                            No change
pkgs/main/noarch                                              No change
pkgs/r/linux-64                                               No change
pkgs/r/noarch                                                 No change
https://conda.software.inl.gov/public/noarch                  No change
https://conda.software.inl.gov/public/linux-64                No change

Pinned packages:
  - python 3.10.*


Transaction

  Prefix: /share/home/ZhangchenFan/mambaforge3/envs/moose

  All requested packages already installed

zc

**Reply by GiudGiud** [reply]
Can you paste the output of ‘mamba list | grep moose’

my guess is that the packages are frozen to an old version due to a dependency resolution issue

**Reply by FanZhangchen** [reply]
The output is here

/share/home/ZhangchenFan/mambaforge3/lib/python3.10/site-packages/conda_package_streaming/package_streaming.py:19: UserWarning: zstandard could not be imported. Running without .conda support.
  warnings.warn("zstandard could not be imported. Running without .conda support.")
/share/home/ZhangchenFan/mambaforge3/lib/python3.10/site-packages/conda_package_handling/api.py:29: UserWarning: Install zstandard Python bindings for .conda support
  _warnings.warn("Install zstandard Python bindings for .conda support")
# packages in environment at /share/home/ZhangchenFan/mambaforge3/envs/moose:
moose-libmesh             2022.11.07              build_2    https://conda.software.inl.gov/public
moose-libmesh-vtk         9.1.0               h7a8feb7_16    https://conda.software.inl.gov/public
moose-mpich               4.0.2                   build_6    https://conda.software.inl.gov/public
moose-petsc               3.16.6                  build_5    https://conda.software.inl.gov/public
moose-test-tools          2023.04.04      py310h0c21752_0    https://conda.software.inl.gov/public
moose-tools               2023.04.04      py310h0c21752_0    https://conda.software.inl.gov/public
moose-wasp                2023.04.04              build_0    https://conda.software.inl.gov/public

**Reply by GiudGiud** [reply]
The packages are a year old

you ll need to remove the moose mamba environment and reinstall it

**Reply by FanZhangchen** [reply]
Thank you. I just want to make sure, should I use the `mamba env remove -n moose` command first and then install the moose environment with the command `mamba create -n moose moose-tools moose-libmesh`?

**Reply by milljm** [reply]
You'll want to be out of the environment before running the remove (Conda won't allow the deletion of the environment it is currently in)

```bash
mamba deactivate
mamba activate base
mamba env remove -n moose
mamba create -n moose moose-tools moose-libmesh
```

In the future, we're hoping to have something in place in our Makefiles that will be able to quickly spot issues like this (an out of date moose-libmesh for example). Also, we're working on a single Conda package installation theme (like `mamba create -n moose moose-dev`).

**Reply by milljm** [reply]
And the stuck issue you have (`mamba update --all` not budging)... I really need to fix that. Unfortunately it means practically deleting the entire /public channel and starting over. The "stuck" issue is all based on how we were pinning packages a year ago (being to strict).

**Reply by FanZhangchen** [reply]
I remove the environment and try to install it. But I found the packages seem to be the same as the ones I removed. I will attach the info

 + moose-libmesh                           2022.11.07  build_0               conda.software.inl.gov/public/linux-64     Cached
  + moose-libmesh-vtk                            9.1.0  h7bcc875_14           conda.software.inl.gov/public/linux-64     Cached
  + moose-mpich                                  4.0.2  build_4               conda.software.inl.gov/public/linux-64     Cached
  + moose-petsc                                 3.16.6  build_3               conda.software.inl.gov/public/linux-64     Cached
  + moose-test-tools                        2023.04.04  py310h0c21752_0       conda.software.inl.gov/public/linux-64     Cached
  + moose-tools                             2023.04.04  py310h0c21752_0       conda.software.inl.gov/public/linux-64     Cached
  + moose-wasp                              2023.04.04  build_0               conda.software.inl.gov/public/linux-64     Cached

**Reply by GiudGiud** [reply]
These are still the old packages. 
As Jason mentioned, delete the enrvironment and the channel and try again.

If that still does not work, delete mamba/miniconda and try again
